File: scripts/052/honma.rb
#!/usr/bin/env ruby
# coding : utf-8
now = Time.now
(123456..166666).each do |n|
  sort_num = n.to_s.split(//).sort
  (2 * n).to_s.split(//).sort == sort_num and
    (3 * n).to_s.split(//).sort == sort_num and
    (4 * n).to_s.split(//).sort == sort_num and
    (5 * n).to_s.split(//).sort == sort_num and
    (6 * n).to_s.split(//).sort == sort_num and
    p n and break
end
p Time.now - now

File: scripts/052/honma.js
for (var i = 123456 ; i <= 166666 ; i++) {
  var sort_num = i.toString().split('').sort().toString();
  if ((2 * i).toString().split('').sort().toString() == sort_num &&
      (3 * i).toString().split('').sort().toString() == sort_num &&
      (4 * i).toString().split('').sort().toString() == sort_num &&
      (5 * i).toString().split('').sort().toString() == sort_num &&
      (6 * i).toString().split('').sort().toString() == sort_num ){
        console.log(i);
        break;
      }
}
